sway-lib-std/src/chain/auth.sw

```diff
@@ -37,24 +33,23 @@ pub fn caller_contract_id() -> ContractId {
     })
 }
 
-/// Get the `Sender` (i.e. `Address` or `ContractId`) from which a call was made.
-/// Returns a `Result::Ok(Sender)`, or `Result::Err(AuthError)` if a sender cannot be determined.
-pub fn msg_sender() -> Result<Sender, AuthError> {
+/// Get the `Identity` (i.e. `Address` or `ContractId`) from which a call was made.
+/// Returns a `Result::Ok(Identity)`, or `Result::Err(AuthError)` if an identity cannot be determined.
+pub fn msg_sender() -> Result<Identity, AuthError> {
     if caller_is_external() {
-        let sender_res = get_coins_owner();
-        match sender_res {
-            Result::Ok(sender) => Result::Ok(sender),
-            sender_res => sender_res,
+        let owner_res = get_coins_owner();
+        match owner_res {
+            Result::Ok(owner) => Result::Ok(owner), owner_res => owner_res, 
```

Comment:
You can just return `owner_res` here directly, no? Saves an object creation.
```suggestion
            Result::Ok(owner) => owner_res, owner_res => owner_res, 
```

Reply:
You could remove the `match` expression entirely, if both arms are returning the same thing, and that thing is what is being matched...

Reply:
The changes to `msg_sender` were introduced in commit 307d91cb1a0260027c5256b9086254a7b47eb189, which is not part of this PR.
That said, there were a few issues last time we tried to clean this up.
- `let owner_res = get_coins_owner();` this call to `get_coins_owner` returns a result which looks like it should be able to be directly returned by `msg_sender`, but they're actually different types (or they were... I'll have to retest this).
- the reason we have `Result::Ok(owner) => Result::Ok(owner)` is because the first `Result` is from `get_coins_owner()` so we can't return it directly. So we extract the inner `owner` value and cast it as a `Result` of the correct type. Will check this out tomorrow and retest to see what might have changed.
If possible I'll remove the `match` entirely. It looks like it should work...

Reply:
OK, we need first class functions so we can add `map` and `map_err` etc.  Carry on.

Reply:
I stand corrected. It seems I was able to successfully remove the `match` altogether. This is only surprising because previously I was not able to do this. Nice to see things progressing on this front!